lib/logger/lib/logger/backends/console.ex

```diff
@@ -97,4 +143,41 @@ defmodule Logger.Backends.Console do
   defp color_event(data, level, %{enabled: true} = colors) do
     [IO.ANSI.format_fragment(Map.fetch!(colors, level), true), data | IO.ANSI.reset]
   end
+
+  defp flush_buffer(%{batch_size: 0} = state) do
+    state
+  end
+
+  defp flush_buffer(%{device: device, buffer: buffer} = state) do
+    next_ref = io_request(device, [buffer | dropping(state)])
+    %{state | ref: next_ref, buffer: [], batch_size: 0}
+  end
+
+  defp dropping(%{batch_size: batch_size, max_buffer: max_buffer})
+  when batch_size <= max_buffer do
+    []
+  end
+
+  defp dropping(state) do
+    %{batch_size: batch_size, max_buffer: max_buffer, colors: colors} = state
+    dropped = batch_size - max_buffer
+    msg = "#{inspect __MODULE__} dropped #{inspect dropped} events as it " <>
+          "exceeded the max buffer size of #{inspect max_buffer} messages"
+    color_event(msg, :error, colors)
+  end
+
+  defp handle_error({:put_chars, :unicode, data} = error, state) do
+    %{device: device} = state
+    case :unicode.characters_to_binary(data) do
```

Comment:
Sorry, what I mean is that we can either do unicode+prune, which you are doing today, or call inspect on the data, which I was doing in another PR.

> Ah I took this approach to prevent an infinite loop, we would never get the binary result on a well behaved IO device.

Ah, I think we should spawn a new process just in case. That's why I called spawn in my original PR but I forgot about it. :P We shouldn't care about replies on this case.